resources/articles/oauth/token.md

```diff
@@ -0,0 +1,199 @@
+```
+
+### Response
+
+```js
+HTTP/1.1 200 OK
+Content-Length: 2034
+Content-Type: application/json
+Date: Mon, 29 Feb 2016 13:37:00 GMT
+
+{
+    "expires_in": 900,
+    "scope": "openid profile",
+    "token_type": "Bearer",
+    "access_token": "eyJ0eXAiOi...Q5MffoMPvA",
+}
+```
+
+## Grant type Password, for user tokens
+
+This grant is mainly only used from native mobile apps, and is not permitted
+in any other application.
+
+### Request
+
+```
+POST /oauth/token
+Authorization: Basic NGU4NDYz...OmZvb2Jhcg==
+Content-Type: application/x-www-form-urlencoded
+
+grant_type=password&scope=openid%20profile&username=john.doe%40example.com&password=qwerty
+```
+
+### Response
+
+```js
+HTTP/1.1 200 OK
+Content-Length: 2034
+Content-Type: application/json
+Date: Mon, 29 Feb 2016 13:37:00 GMT
+
+{
+    "expires_in": 3600,
+    "scope": "openid profile",
+    "token_type": "Bearer",
+    "access_token": "eyJ0eXAi...sVAfIcA0",
+    "refresh_token": "eyJ0eXAi...nom5mDrl-k",
+    "id_token": "eyJ0eXAi...rEXE05y1Cg",
+}
+```
+
+## Grant type Authorization code, for user tokens
+
+The authorization code grant is the second step in the authorization flow for user.
+A user logs in to SPiD, and SPiD will return user to site with a code.
+That code is then exchanged in backend for a user token.
+
+### Request
+
+```
+POST /oauth/token
+Authorization: Basic NGU4NDYz...OmZvb2Jhcg==
+Content-Type: application/x-www-form-urlencoded
+
+grant_type=authorization_code&code=eyJ0eXAi...sVAfIcA0&redirect_uri=https%3A%2F%2Fexample.com
+```
+
+### Response
+
+```js
+HTTP/1.1 200 OK
+Content-Length: 2034
+Content-Type: application/json
+Date: Mon, 29 Feb 2016 13:37:00 GMT
+
+{
+    "expires_in": 3600,
+    "scope": "openid profile",
+    "token_type": "Bearer",
+    "access_token": "eyJ0eXAi...sVAfIcA0",
+    "refresh_token": "eyJ0eXAi...nom5mDrl-k",
+    "id_token": "eyJ0eXAi...rEXE05y1Cg",
+}
+```
+
+## Grant type Refresh token, for user tokens
+
+The refresh token grant is used to fetch a new user access token, when the
+current access token has expired. Access token lifetimes are typically shorter,
+and refresh token lifetimes are typically longer.
+
+### Request
+
+```
+POST /oauth/token
+Authorization: Basic NGU4NDYz...OmZvb2Jhcg==
+Content-Type: application/x-www-form-urlencoded
+
+grant_type=refresh_token&refresh_token=eyJ0eXAi...nom5mDrl-k&scope=openid%20profile
+```
+
+### Response
+
+```js
+HTTP/1.1 200 OK
+Content-Length: 2034
+Content-Type: application/json
+Date: Mon, 29 Feb 2016 13:37:00 GMT
+
+{
+    "expires_in": 3600,
+    "scope": "openid profile",
+    "token_type": "Bearer",
+    "access_token": "eyJ0eXAi...sVAfIcA1",
+    "refresh_token": "eyJ0eXAi...nom5mDr1Cr",
+    "id_token": "eyJ0eXAi...rEXE05y1Cg",
+}
+```
+
+## Grant type JWT Bearer, for user tokens via Facebook and Google+
+
+The JWT Bearer grant is used to fetch a user access token, by first logging
+the user in via a third party, and then sending third party token to SPiD,
```

Comment:
"sending third party" -> "sending the third party"

Reply:
🥇